gitlab/client.py

```diff
@@ -779,10 +779,20 @@ def http_request(
             if 200 <= result.status_code < 300:
                 return result
 
-            if (429 == result.status_code and obey_rate_limit) or (
-                result.status_code in gitlab.const.RETRYABLE_TRANSIENT_ERROR_CODES
-                and retry_transient_errors
-            ):
+            def should_retry() -> bool:
+                if result.status_code == 429 and obey_rate_limit:
```

Comment:
Probably a nit pick but I would suggest using the `Any` function. Not a show stopper for this PR though.

Reply:
Not sure I understand here - did you mean `all()`?

The previous implementation with the 409 condition added was too complex for pylint so I split into a more explicit helper with separate if branches.

Reply:
Well, it could be a combination of `all` and `any`.  I get that the result my be too complex for pylint and for maintenance in general.

Reply:
Yes, I think for now this might be the most readable option. Maybe the match statement once we're on python 3.10 will make life a bit easier :) Thanks!